support/ci/shared.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -euo pipefail
+
+get_current_toolchain() {
+  # It turns out that every nightly version of rustfmt has slight tweaks from the previous version.
+  # This means that if we're always using the latest version, then we're going to have enormous
+  # churn. Even PRs that don't touch rust code will likely fail CI, since master will have been
+  # formatted with a different version than is running in CI. Because of this, we're going to pin
+  # the version of nightly that's used to run rustfmt and bump it when we do a new release.
+  #
+  # Note that not every nightly version of rust includes rustfmt. Sometimes changes are made that
+  # break the way rustfmt uses rustc. Therefore, before updating the pin below, double check
+  # that the nightly version you're going to update it to includes rustfmt. You can do that
+  # using https://mexus.github.io/rustup-components-history/x86_64-unknown-linux-gnu.html
+  echo "nightly-2019-03-04"
```

Comment:
A couple of questions - can you confirm:
1. I assume we will need to also make sure dev environments have this specific nightly toolchain installed?
2. VSCode/Atom/RLS/etc plug-ins can pick up the rustfmt options and format-on-save automatically as today?
3. Builds will continue to use stable, not nightly toolchain?

Reply:
1. Correct
1. Assuming your editor in question allows you to customize the command that's used to run `rustfmt`, correct. I can confirm this is the case for vim and emacs. I have no experience with the others. For example, for vim, I added this line to my `~/.vimrc`: `let g:rustfmt_command = 'rustup run nightly-2019-03-04 rustfmt'`
1. Correct

Reply:
Thanks... one more q: there are no actual formatting changes in this PR based on the new settings?

Reply:
Yeah, I was waiting to see if I hooked up CI right before adding those changes.  I think it's good though.  I'll add them now.